src/firstboot/firstboot.c

```diff
@@ -638,7 +679,7 @@ static int write_root_passwd(const char *passwd_path, const char *password) {
                         .pw_gid = 0,
                         .pw_gecos = (char *) "Super User",
                         .pw_dir = (char *) "/root",
-                        .pw_shell = (char *) "/bin/sh",
+                        .pw_shell = shell ? (char *) shell : (char *) "/bin/sh",
```

Comment:
`(char*) (shell ?: "/bin/sh")` ?

Reply:
Nice one! I keep forgetting about all this GCC goodness.